src/include/gurt/common.h

```diff
@@ -265,11 +265,26 @@ char *d_realpath(const char *path, char *resolved_path);
 #define D_REALLOC_ARRAY_Z(newptr, oldptr, count)			\
 	D_REALLOC_COMMON(newptr, oldptr, 0, sizeof(*(oldptr)), count)
 
-#define D_FREE(ptr)							\
-	do {								\
-		D_DEBUG(DB_MEM, "free '" #ptr "' at %p.\n", (ptr));	\
-		d_free(ptr);						\
-		(ptr) = NULL;						\
+/* Free a pointer.  Always logs as this both records that the memory has been freed but also shows
```

Comment:
What is NLF?  No log free ?  Technically, we are logging them, just not if they are NULL.  Is there a benefit to ever logging them when they are NULL?

Reply:
> What is NLF? No log free ? Technically, we are logging them, just not if they are NULL.

Yes, "No Log Free".  Good point, it's the wrong name.

> Is there a benefit to ever logging them when they are NULL?

Not a massive one.

I'm happy with either this or remove logging of all NULL values, what I'm trying to avoid is a few locations spewing 100s of zero-value lines into log files.